modules/syncMinimongo.js

```diff
@@ -100,34 +90,30 @@ exports.backendSync = function() {
 
 
 exports.frontendSync = function(coll) {
-    var ipc = electron.ipcRenderer;
+    var ipc = electron.ipcRenderer,
+        collName = coll._name,
+        syncDoneResolver;
 
-    var collName = coll._name;
-
-    console.debug('Reload collection from backend: ', collName);
-
-    var syncDoneResolver = null;
     coll.onceSynced = new Promise(function(resolve, reject) {
         syncDoneResolver = resolve;
     });
 
-
     (new Promise(function(resolve, reject) {
-        var dataStr = ipc.sendSync('minimongo-reloadSync', {
+        var dataStr, dataJson;
+
+        dataStr = ipc.sendSync('minimongo-reloadSync', {
             collName: collName
         });
 
-        if (!dataStr) {
+        if (!dataStr || (dataJson = JSON.parse(dataStr)).length == 0) {
```

Comment:
JSON.parse() needs to go inside the `try` block below where it originally was in case it throws an Error.